babelsberg/tests.js

```diff
@@ -2340,5 +2423,94 @@ TestCase.subclass('users.timfelgentreff.babelsberg.tests.AutomaticSolverSelectio
         this.assert(man.shirt !== man.pants, "shirt and pants must not have the same color");
         this.assert(man.pants === "black" || man.pants === "blue" || man.pants === "white", "pants should be 'black', 'blue' or 'white'");
     },
+
+    testReevaluationAfterDefaultNumberOfSolvingOperations: function() {
+        preparePatchedSolvers();
+        var obj = {a: 2, b: 3};
+        bbb.defaultSolvers[0].forcedDelay = 10;
+        bbb.defaultSolvers[1].forcedDelay = 0;
+        bbb.defaultRecalculationInterval = 2; // recalculate after two updates
+        var constraint = bbb.always({
+            ctx: {
+                obj: obj
+            }
+        }, function() {
+            return obj.a + obj.b == 3;
+        });
+        this.assert(constraint.solver === bbb.defaultSolvers[1],
+                    "the initially faster solver should have been chosen");
+        bbb.defaultSolvers[0].forcedDelay = 0;
+        bbb.defaultSolvers[1].forcedDelay = 10;
+        for (var i = 0; i < 2; i++) {
+            // BUG with variable changes feature:
+            // the definingSolver for CV a is wrong, because the check
+            // whether there is an enabled constraint is commented out
+            // in CV._searchDefiningSolver
+            obj.a += 1;
```

Comment:
Now that the check in searchDefiningSolver is back in, this comment can be deleted. I just wrote it down back then so it is not forgotten and there is a hint if this test turns red after the merge.